great_expectations_provider/example_dags/example_great_expectations_dag.py

```diff
@@ -3,113 +3,208 @@
     data_context_root_dir=ge_root_dir,
+    checkpoint_name="taxi.pass.chk",
     dag=dag,
 )
 
-# This runs a checkpoint that will fail, but we set a flag to exit the task successfully.
-ge_checkpoint_fail_but_continue = GreatExpectationsOperator(
-    task_id="ge_checkpoint_fail_but_continue",
-    run_name="ge_airflow_run",
+ge_data_context_root_dir_with_checkpoint_name_fail_validation_and_not_task = GreatExpectationsOperator(
+    task_id="ge_data_context_root_dir_with_checkpoint_name_fail_validation_and_not_task",
+    data_context_root_dir=ge_root_dir,
     checkpoint_name="taxi.fail.chk",
     fail_task_on_validation_failure=False,
-    data_context_root_dir=ge_root_dir,
     dag=dag,
 )
 
-# This runs a checkpoint and passes in a root dir.
-ge_checkpoint_pass_root_dir = GreatExpectationsOperator(
-    task_id="ge_checkpoint_pass_root_dir",
-    run_name="ge_airflow_run",
-    checkpoint_name="taxi.pass.chk",
+ge_checkpoint_kwargs_substitute_batch_request_fails_validation_but_not_task = GreatExpectationsOperator(
+    task_id="ge_checkpoint_kwargs_substitute_batch_request_fails_validation_but_not_task",
     data_context_root_dir=ge_root_dir,
+    checkpoint_name="taxi.pass.chk",
+    checkpoint_kwargs={"expectation_suite_name": "taxi.demo_fail"},
+    fail_task_on_validation_failure=False,
     dag=dag,
 )
 
-# This runs the expectation suite at "gs://$my_bucket/expectations/taxi/demo.json"
-# against the data in the BigQuery table "$my_project.$my_dataset.$my_table", where
-# $my_bucket, $my_project, $my_dataset, and $my_table are Airflow variables of the
-# same name. The connection named "my_bigquery_conn_id" is used to connect to GCP.
-ge_bigquery_validation = GreatExpectationsBigQueryOperator(
-    task_id="ge_bigquery_validation",
-    gcp_project="{{ var.value.my_project }}",
-    gcs_bucket="{{ var.value.my_bucket }}",
-    # GE will use a folder "$my_bucket/expectations"
-    gcs_expectations_prefix="expectations",
-    # GE will use a folder "$my_bucket/validations"
-    gcs_validations_prefix="validations",
-    # GE will use a folder "$my_bucket/data_docs"
-    gcs_datadocs_prefix="data_docs",
-    # GE will look for a file $my_bucket/expectations/taxi/demo.json
-    expectation_suite_name="taxi.demo",
-    table="{{ var.value.my_table }}",
-    bq_dataset_name="{{ var.value.my_dataset }}",
-    bigquery_conn_id="my_bigquery_conn_id",
+ge_data_context_config_with_checkpoint_config_pass = GreatExpectationsOperator(
+    task_id="ge_data_context_config_with_checkpoint_config_pass",
+    data_context_config=data_context_config,
+    checkpoint_config=checkpoint_config,
     dag=dag,
 )
 
 
+ge_checkpoint_fails_and_runs_callback = GreatExpectationsOperator(
+    task_id="ge_checkpoint_fails_and_runs_callback",
+    data_context_root_dir=ge_root_dir,
+    checkpoint_name="taxi.fail.chk",
+    fail_task_on_validation_failure=False,
+    validation_failure_callback=(lambda x: print("Callback successfully run", x)),
```

Comment:
This is really cool! Love that this functionality is being included in the example.